app-backend/api/src/main/scala/shape/Routes.scala

```diff
@@ -44,12 +47,63 @@ trait ShapeRoutes extends Authentication
       get { listShapes } ~
       post { createShape }
     } ~
-    pathPrefix(JavaUUID) { shapeId =>
-      pathEndOrSingleSlash {
-        get { getShape(shapeId) } ~
-        put { updateShape(shapeId) } ~
-        delete { deleteShape(shapeId) }
+      post {
+        pathPrefix("upload") {
+          pathEndOrSingleSlash {
+            authenticate { user =>
+              val tempFile = ScalaFile.newTemporaryFile()
```

Comment:
these files won't delete themselves if the jvm crashes. call `deleteOnExit()` on the file to prevent this.
You can also use temp files either in a for comprehension:
```
for {
 tempFile <- File.temporaryFile()
} doSomething(tempFile) // tempFile is auto deleted at the end of this block - even if an exception happens
```
or equivalently
```
File.usingTemporaryFile() {tempFile =>
  //do something
}  // tempFile is auto deleted at the end of this block - even if an exception happens
````

Reply:
We also don't seem to be checking file sizes - I doubt we want to accept shapefiles that are a gigabyte in size this way, so smart limit might be something like 20mb in nginx

Reply:
We should silently simplify polygons to a reasonable point limit.

Reply:
Limited the max upload size in nginx with https://github.com/raster-foundry/raster-foundry/pull/2919/commits/55cf50136df9814829145905bb04d1675e300e8b

There isn't really a concept of simplifying to a number of points -- we can do some simplification with tolerances, but I'm not sure there is a one-size-fits-all approach here. There is a fair amount of experimentation we can do, but I don't think we'll have a good sense until we see some usage.